lib/index.js

```diff
@@ -83,6 +83,7 @@ module.exports = {
         (err, data) => {
           if (err) return reject(err);
           file.url = converter.getUrl(data);
+          delete file.buffer;
```

Comment:
I think the issue comes from line 99 and not this one.
Strapi uses `uploadStream` (using streams) if it exists, otherwise `upload` (using buffers).
When Strapi uses `upload`, Strapi adds the buffer to the file before calling `upload` and removes it just after.
So I think this change won't have any impact. Also I think it's better to let the responsability to remove it to Strapi.

I see that the `uploadStream` is not really using the streams as it's converting it to buffer. There are some issues with Buffers in general (memory leaks), that's why Strapi moved forward streams instead.
I would say that the best fix I can see is to re-write a bit the provider to follow the same logic as the `aws` one : https://github.com/strapi/strapi/blob/main/packages/providers/upload-aws-s3/lib/index.js

Reply:
Here `uploadStream` uses `upload` under the hood and strapi never removes the buffer (since it assumes it uses a stream). I simply added the deletion to line 86 since it requires the least code changes (and from my understanding would remove the buffer after `upload` is called by `uplaodStream`, but it is not unlikely that I overlooked something).

I agree that the uploadStream implementation would benefit from a overhaul like suggested.

Reply:
I created the issue #11 to keep track of this. Feel free to close this PR if another approach is taken.

Reply:
Yes you're right I forgot it was called by `uploadStream`. I would advise to delete the buffer in `uploadStream` however, in order to not mix the usercases and to not change to much things. However I still recommand to rewrite the provider as describe in my first comment

Reply:
@petersg83 I totally agree. This provider could be improved by using the same pattern as the S3 provider, therefore making full use of the stream instead of concatenating. I worry about the potential breaking changes but I have doubts if there will be any.

Feel free to apply the changes you advised in this PR.